### PR TITLE
Node installation and translations

### PR DESCRIPTION
**Pull request by giacomoorsi**
By now we have implemented 11ty with its language extensions. Ideally, we would like to have openritardi.it in Italian, and then the English translation on the `/en/*` subfolder. This is currently achieved by manually specifying the permalink in each page in the `src/en` and `src/it` folders (seems like a good hack!)

### How to generate the new website: 
First, unzip the data with the new script
```
bash prepare_data.sh
```
Then use `npm` to compile the website
```bash
cd website
npm install
npm run build
```

The folder `dist` is the entrypoint of the website. 

### TODO: 

- [x] Make a common header menu on a single file, and loading it from all templates. The tricky part is on how to assign the class `uk-active` to the menu item of the shown webpage. This can be done probably using the variable `{{ page.url }}` in an if-statement
- [x] Translate all labels contained in the HTML templates (those in `_include/*.njk` and put them into `translations.json` (perhaps we can create a different file for each page?)
- [ ] Translate all labels contained in all JS files (figure out a way to do those, because they're trickier, since right now all the JS files are in the `static` folder). This is on hold waiting for progress on #6 which is refactoring javascript code by adding html templates

### CODE REVIEW

File: website/src/_data/customFilters.js
Comment (hadriensevel) [inline website/src/_data/customFilters.js:1]: File name should be "custom-filters.js" ([see Google JavaScript Style Guide](https://google.github.io/styleguide/jsguide.html#file-name))

File: website/src/_data/customFilters.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+var nunjucks = require('nunjucks');
```
Comment (hadriensevel) [inline website/src/_data/customFilters.js:1]: No "var"! 90% of the time it should be "const" ([see Google JavaScript Style Guide](https://google.github.io/styleguide/jsguide.html#features-use-const-and-let))

File: website/package.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "openritardi",
+    "version": "1.0.0",
+    "scripts": {
+        "build": "npx @11ty/eleventy"
+    },
+    "dependencies": {
```
Comment (hadriensevel) [inline website/package.json:7]: It should be "devDependencies" since the dependencies are only during development and not during production

**Comment by giacomoorsi** [comment]
Considering merging even if last bullet is not implemented yet, in order to be able to progress with other website developments on the new NodeJS website

**Comment by giacomoorsi** [comment]
Merging even if last bullet not implemented yet. This means that the labels provided by Javascript scripts will not be translated in Italian yet, but at least we have the rest of the website.

**Comment by Efois** [comment]
> By now we have implemented 11ty with its language extensions. Ideally, we would like to have openritardi.it in Italian, and then the English translation on the `/en/*` subfolder. This is currently achieved by manually specifying the permalink in each page in the `src/en` and `src/it` folders (seems like a good hack!)
> 
> ### How to generate the new website:
> First, unzip the data with the new script
> 
> ```
> bash prepare_data.sh
> ```
> 
> Then use `npm` to compile the website
> 
> ```shell
> cd website
> npm install
> npm run build
> ```
> 
> The folder `dist` is the entrypoint of the website.
> 
> ### TODO:
> * [x]  Make a common header menu on a single file, and loading it from all templates. The tricky part is on how to assign the class `uk-active` to the menu item of the shown webpage. This can be done probably using the variable `{{ page.url }}` in an if-statement
> * [x]  Translate all labels contained in the HTML templates (those in `_include/*.njk` and put them into `translations.json` (perhaps we can create a different file for each page?)
> * [ ]  Translate all labels contained in all JS files (figure out a way to do those, because they're trickier, since right now all the JS files are in the `static` folder). This is on hold waiting for progress on [Refactor Javascript and add Rollup bundler #6](https://github.com/giacomoorsi/OpenRitardi/pull/6) which is refactoring javascript code by adding html templates

Could you need help in translating or reviewing labels in italian language?

**Comment by giacomoorsi** [comment]
> Could you need help in translating or reviewing labels in italian language?

Hi @Efois that would be awesome: we mainly need help changing the logic of the website as documented here #16 rather than only translating the labels from English to Italian

**Pull request merged**